### PR TITLE
feature: Support Arduino in Wokwi

### PR DESCRIPTION
**Pull request by P-R-O-C-H-Y**
Adds support for pytest-embedded-wokwi to run with Arduino build files.

### CODE REVIEW

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py [hunk 1]
```diff
@@ -101,9 +102,16 @@ def create_wokwi_toml(self):
 
     def create_diagram_json(self):
         app = self.app
-        diagram_json_path = os.path.join(app.app_path, 'diagram.json')
         target_board = target_to_board[app.target]
 
+        # Check for specific target.diagram.json file first
+        diagram_json_path = os.path.join(app.app_path, (app.target + '.diagram.json'))
+        if os.path.exists(diagram_json_path):
+            shutil.copyfile(diagram_json_path, os.path.join(app.app_path, 'diagram.json'))
```
Comment (hfudev) [inline pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:110]: here we're copying the `esp32.diagram.json` to `diagram.json`. what will happen if user has both files? the `diagram.json` will be overwritten.

does wokwi support setting the diagram path?
- if yes, we could use `wokwi --diagram-filepath esp32.diagram.json` directly
- if no, we should backup user's `diagram.json` to `diagram.json.old` first (when exist), then do the copy

**Review by hfudev — CHANGES_REQUESTED** [review]

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py (hunk 1 above)
Comment (P-R-O-C-H-Y) [reply pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:110]: I will check and update accordingly.

File: pytest-embedded-wokwi/tests/test_wokwi.py [hunk 2]
```diff
@@ -36,3 +36,22 @@ def test_pexpect_by_wokwi(dut):
     )
 
     result.assert_outcomes(passed=1)
+
+
+def test_pexpect_by_wokwi_esp32_arduino(testdir):
```
Comment (hfudev) [inline pytest-embedded-wokwi/tests/test_wokwi.py:41]:
```suggestion
@wokwi_cli_required
@wokwi_token_required
def test_pexpect_by_wokwi_esp32_arduino(testdir):
```

**Review by hfudev — CHANGES_REQUESTED** [review]

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py (hunk 1 above)
Comment (P-R-O-C-H-Y) [reply pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:110]: There is no argument to specify diagram.json file path.
So I will backup the diagram.json to diagram.json.old. Do we need to restore it afterwards or check also for the .old one?

File: pytest-embedded-wokwi/tests/test_wokwi.py (hunk 2 above)
Comment (P-R-O-C-H-Y) [reply pytest-embedded-wokwi/tests/test_wokwi.py:43]: Done

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py (hunk 1 above)
Comment (P-R-O-C-H-Y) [reply pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:117]: I think the main usage will be just single diagram.json or multiple of them specifying it for each target.
I dont think both options together will be used, it doesn't make much sense.

File: pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py [hunk 3]
```diff
@@ -101,9 +102,23 @@ def create_wokwi_toml(self):
...
         app = self.app
-        diagram_json_path = os.path.join(app.app_path, 'diagram.json')
         target_board = target_to_board[app.target]
 
+        # Check for specific target.diagram.json file first
+        diagram_json_path = os.path.join(app.app_path, (app.target + '.diagram.json'))
+        if os.path.exists(diagram_json_path):
+            # If there is also common diagram.json file, backup it first to diagram.json.old
+            if os.path.exists(os.path.join(app.app_path, 'diagram.json')):
+                shutil.copyfile(
```
Comment (hfudev) [inline pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:112]: let's give user some warning here

```python
logging.warning('using %s instead. backup the original diagram.json to diagram.json.old', diagra_json_path)
```

Comment (P-R-O-C-H-Y) [reply pytest-embedded-wokwi/pytest_embedded_wokwi/wokwi_cli.py:115]: Done :)

**Review by hfudev — APPROVED** [review]

**Comment by hfudev** [comment]
Thank you for this PR!

**Pull request merged**